pytensor/tensor/subtensor.py

```diff
@@ -186,8 +219,8 @@ def get_idx_list(inputs, idx_list):
 
 
 def get_canonical_form_slice(
-    theslice: slice | Variable, length: Variable
-) -> tuple[Variable, int]:
+    theslice: slice | Variable, length: ScalarVariable
+) -> tuple[slice | Variable, int | ScalarConstant]:
```

Comment:
This is changing the signature of the function? According to the docstrings this should return a slice and an indicator integer variable -1/1, not a ScalarConstant?